.github/workflows/build.yaml

```diff
@@ -0,0 +1,41 @@
+name: Build target user
+
+on: [pull_request, workflow_dispatch]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python 3.x
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.x'
+    - name: Copy source.list file to include deb-src
+      run: |
+        sudo cp /etc/apt/sources.list /etc/apt/sources.list.d/tmp.list
+        sudo sed -i "s/# deb-src/deb-src/g" /etc/apt/sources.list.d/tmp.list
+    - name: Install deps
+      run: |
+        sudo apt-get  update
+        sudo apt-get --no-install-recommends -y build-dep qemu
+        sudo apt-get install -y autoconf libtool protobuf-c-compiler
+        pip3 install --user ninja
+    - name: Install OCaml
+      uses: ocaml/setup-ocaml@v2
+      with:
+        ocaml-compiler: 4.14.x
+        dune-cache: true
+        opam-disable-sandboxing: true
+    - name: Install piqi
+      run: |
+        opam install piqi
+    - name: Clone qemu and bap-frames
+      run: |
+        git clone --depth 1 http://github.com/BinaryAnalysisPlatform/bap-frames.git
+        git clone --depth 1 http://github.com/BinaryAnalysisPlatform/qemu.git
+    - name: Build without tracewrap
+      run: |
+        cd qemu
+        ./configure --prefix=$HOME --target-list=arm-linux-user
```

Comment:
We also want to build the code guarded by the `HAS_TRACEWRAP` variable.

Reply:
Sorry, I think we talk passed each other.

The code guarded in `HAS_TRACEWRAP` is never used in this branch. This is kinda the idea. To have only a base branch which has the tracewrap files without actually using them. So people can base their target trase branches ontop of this one, without relying on other archs. The building of the tracewrap code gets tested in each separated `tracewrap-8.1-<target>` branch.

Reply:
> The code guarded in HAS_TRACEWRAP is never used in this branch.

But I am seeing a few `ifdef HAS_TRACEWRAP`-guarded pieces of code in the diff. So if it is not set it won't be built.

Reply:
Yes, exactly. The code wrapped in `#ifdef HAS_TRACEWRAP` is our trace code (no original QEMU code is wrapped in it) and should only be build if a target uses it.
But this branch has no target implemented.

When I open the PR for the Hexagon tracing based on `tracewrap-8.1` I'll add an CI job to build with tracewrap and for `linux-user-hexagon`. With it the code wrapped in `#ifdef HAS_TRACEWRAP` gets build.

This is the branching strategy described here: https://github.com/BinaryAnalysisPlatform/qemu/issues/31

This PR is supposed to be the `tracewrap-8.1-base` branch.